app/services/phone_recaptcha_mock_validator.rb

```diff
@@ -0,0 +1,14 @@
+class PhoneRecaptchaMockValidator < PhoneRecaptchaValidator
```

Comment:
Since the NewPhoneForm is already doing a check to see which class it uses, what if we just had it pass in (dependency injection style) the mock Recaptcha validator?

Reply:
Was your idea to pass the internal `validator` delegatee as a constructor argument to `PhoneRecaptchaValidator`? One potential complication there is that `score_threshold` is a method of `PhoneRecaptchaValidator` itself, so not directly accessible from `NewPhoneForm`. I'll poke around at this a bit more though. Maybe we can pass the class reference and have it referenced in the `PhoneRecaptchaValidator#validator` method.

Example:

```rb
def validator
  validator_class.new(**validator_args)
end
```

Reply:
My thinking was more like this, that all the branching/variable behavior is wired up in one spot? And the goal would be to only create mock classes of things that need to be mocked, because ideally we could wire in the rest through config.

But looking at it, it's prone to different misconfiguration errors.

Maybe if we liked that goal, it's a reason to not have a PhoneRecaptchaValidator as well as a plain RecaptchaValidator because we have to mock each version? Like since we only use Recaptcha in one flow in the app, we collapse those classes back?

```ruby
# new_phone_form.rb
def recaptcha_validator
  @recaptcha_validator ||= begin
    PhoneRecaptchaValidator.new(
      parsed_phone:,
      recaptcha_version:,
      analytics:,
      recaptcha_validator: IdentityConfig.store.phone_recaptcha_mock_validator ?
        RecaptchaMockValidator.new(score:) :
        RecaptchaValidator.new,
    )
  end
end

Reply:
How do you feel about the approach in 011d81d5ab0628503cc9de4df172ff6da6098007 ? It gets rid of the extra class and ends up being a bit more succinct overall.

Reply:
I like it! feels like what I was trying to get at here, and one less class is a huge win IMO